code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -160,6 +179,11 @@
 	. = ..()
 
 /mob/living/simple_animal/Life()
+	if(is_aquatic && !submerged() && stat != DEAD)
+		walk(src, 0)
+		if(!HAS_STATUS(src, STAT_PARA)) // gated to avoid redundant update_icon() calls.
```

Comment:
I'll note that this probably means you'll get a free tick or two without paralysis every 3 or so ticks, but maybe that's intended?